cumulus_library/log_utils.py

```diff
@@ -0,0 +1,95 @@
+"""A set of convenience functions for database logging"""
+
+from cumulus_library import (
+    __version__,
+    base_utils,
+    databases,
+    enums,
+    errors,
+    study_parser,
+)
+from cumulus_library.template_sql import base_templates, sql_utils
+
+
+def log_transaction(
+    *,
+    cursor: databases.DatabaseCursor,
+    schema: str,
+    manifest: study_parser.StudyManifestParser,
+    status: enums.LogStatuses | str | None = enums.LogStatuses.INFO,
+    message: str | None = None,
+):
+    if isinstance(status, str):
+        try:
+            status = enums.LogStatuses(status)
+        except ValueError as e:
+            raise errors.CumulusLibraryError(
+                f"Invalid event type {status} requested for transaction log.\n"
+                f"Valid types: {','.join([x.value for x in enums.LogStatuses])}"
+            ) from e
+    _log_table(
+        table=sql_utils.TransactionsTable(),
+        cursor=cursor,
+        schema=schema,
+        manifest=manifest,
+        dataset=[
+            [
+                manifest.get_study_prefix(),
+                __version__,
+                status.value,
+                base_utils.get_utc_datetime(),
+                message if message else None,
```

Comment:
nit: I think `message or None` is maybe slightly more readable, but if/else is good too.